types/jest/index.d.ts

```diff
@@ -804,7 +804,7 @@ declare namespace jest {
      *  myApi.myApiMethod.mockImplementation(() => "test");
      */
     type Mocked<T> = {
-        [P in keyof T]: T[P] & MockInstance<T[P], ArgsType<T[P]>>;
+        [P in keyof T]: T[P] extends (...args: any[]) => any ? MockInstance<ReturnType<T[P]>, ArgsType<T[P]>>: T[P];
```

Comment:
Why don't you return to `MockInstance<F>` and derive ReturnType and ArgsType where needed? This is a breaking change anyway - so this shouldn't be a problem. Same is true for `Mock<F>` and `jest.fn<F>`. This would make the API less error prone/more obvious as I proposed in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/32864/files#r254967801.

Reply:
this will be discussed in #32901